### DISCUSSION TITLE
Reading ASN1 Content

### DISCUSSION THREAD

**Discussion by RaduBogdan98** [bcgit/bc-java#1341] [Q&A] [answered]
Hello community!
Here is my issue:
I am trying to read ASN.1 content from a file. My file content looks like this:

My-Schema DEFINITIONS IMPLICIT TAGS ::= 
BEGIN
  MySequence ::= SEQUENCE       
  {                                                     
     str1      OCTET STRING (SIZE(1..50)),
     str2      OCTET STRING (SIZE(1..50)),
     str3      OCTET STRING (SIZE(1..50)),
     str4      OCTET STRING (SIZE(1..50)),
     str5      OCTET STRING (SIZE(1..50))
   }                                                     
END

I am doing this using the ASN1InputStream class, however I am not able to do it. When reading an object from my stream I always get DLApplicationSpecific and no useful content.

Here is my code:
      try {
            // Open the .asn1 file using a FileInputStream
            FileInputStream inputStream = new FileInputStream("input.asn1");

            // Create an instance of ASN1InputStream using the input stream
            ASN1InputStream asnInputStream = new ASN1InputStream(inputStream);

            // Read the ASN.1 structure from the input stream
            ASN1Primitive asn1Structure = asnInputStream.readObject();

            inputStream.close();
        } catch (IOException e) {
            e.printStackTrace();
        }

What am I doing wrong and how can I manage to read this input correctly?
The Bouncy Castle version I use is 1.70 with jdk 15 on

**Comment by peterdettman** [comment]
We could help if you provide an example input file. Most likely the file is not a direct BER encoding but has been Hex- or Base64- encoded in some fashion.

**Comment by RaduBogdan98** [comment]
Hi, so my input file would be this:
[input.txt](https://github.com/bcgit/bc-java/files/10869813/input.txt)
It is just dummy input, but the idea is that I need to read files in this asn.1 notation, in order to have the posibility to create/edit the file manually. I may define it wrong, please correct me if that is the case.

Do I need to encode it in order to read it by this ASN1InputStream?

**Comment by peterdettman** [comment]
The ASN1InputStream is for decoding BER encodings. We don't have parsers for ASN.1 module definitions (i.e. abstract syntax).

**Comment by RaduBogdan98** [comment]
And can I encode this ASN.1 notation to BER using your libraries?

**Comment by dghgit** [comment, marked as answer]
You'd need to build the actual structure in Java - BC has the support to allow you to describe the generation and encoding of the above in Java. It's not an ASN.1 language parser though, it just supports parsing of the encodings.
